File: Function/Assets/XLuaFrame/Lua/Message/MgsBase.lua

local MgsBase = {}


--定义消息对象
function MgsBase:new(mgsId)

    local o = {};
    setmetatable(o, self)
    self.__index = self
    o.mgsId = mgsId;
    return o;

end
--获取对应的管理器
function MgsBase:GetManager()
    
    --识别是哪个管理器
    local id = math.floor(  self.mgsId / 3000 ) * 3000;
    id = math.floor( id );

    for i,v in pairs(ManageId) do 
        if id == v then
            return v;
        end
    end
    return nil;
end

return MgsBase;

File: Function/Assets/SimpleFrame/Scripts/Message/MgsBase.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class MgsBase  {

    //16位，4个字节，65536个信息，优化节点。
    public ushort mgsId;

    public MgsBase(ushort mgsId){

        this.mgsId = mgsId;
    }

    //获取相应的管理器
    public ManagerId GetManager(){

        int index = mgsId / FrameTool.mgsId;
        return (ManagerId)(index*FrameTool.mgsId);
    }

    //设置标识
    public void SetMgsId(ushort mgsId){

        this.mgsId = mgsId;
    }


}
